feat: 충전소와 충전기를 업데이트와 추가하는 기능을 구현한다

sosow0212 opened the pull request:
## 📄 Summary
> 충전소와 충전기가 변경 되었다면 업데이트를 하고, 추가 되었다면 새로 추가하는 기능을 만들었습니다.
아직 주기적으로 동작하는 기능은 없습니다.

## 🙋🏻 More
> 


close #26

sosow0212 committed:
feat: 업데이트 됐는지 확인하는 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
feat: 업데이트 됐는지 확인하는 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
test: 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
test: 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
feat: 충전기 및 충전소 업데이트 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
feat: 충전기 및 충전소 업데이트 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/charger/Charger.java:
```diff
@@ -49,17 +51,37 @@ public class Charger {
         return chargerStatus.isAvailable();
     }
+
+    public boolean isUpdated(final Charger charger) {
```
final 제거해주세요~

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/charger/ChargerStatus.java:
```diff
@@ -35,4 +35,5 @@ public class ChargerStatus {
     public boolean isAvailable() {
         return chargerState.isStandBy();
     }
+
```
개행 제거해주세요~

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/ChargeStation.java:
```diff
@@ -64,15 +64,62 @@ public int getAvailableCount() {
+    }
+
+    @Override
+    public boolean equals(final Object o) {
```
final 제거해주세요~

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/ChargeStation.java:
```diff
@@ -64,15 +64,62 @@ public int getAvailableCount() {
+        this.chargers = chargers;
+    }
+
+    public boolean isUpdated(final ChargeStation station) {
```
ditto

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/ChargeStation.java:
```diff
@@ -64,15 +64,62 @@ public int getAvailableCount() {
+
+        if (!latitude.getValue().equals(station.getLatitude().getValue())) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
+
+        if (!longitude.getValue().equals(station.longitude.getValue())) {
+            return true;
+        }
+
```
latitude, longitude 에 equals를 사용해도 좋을 것 같아요!

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/ChargeUpdateJdbc.java:
```diff
@@ -0,0 +1,114 @@
+                .addValue("stationId", charger.getStationId())
+                .addValue("chargerId", charger.getChargerId())
+                .addValue("type", charger.getType())
+                .addValue("address", charger.getAddress())
```
charger에 address가 또 들어간 것 같아요..,,

sosow0212 replied:
정말 컨플릭트 미쳐버리겠네요.. 🥲

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/service/chargerstation/UpdateScrapper.java:
```diff
@@ -0,0 +1,66 @@
+@RequiredArgsConstructor
+@Service
+public class UpdateScrapper {
+    private static final String REQUEST_URL = "/getChargerInfo";
+    private static final int ROW_SIZE = 9999;
+    private static final String DATA_TYPE = "JSON";
+    private final RestTemplate restTemplate;
+    private static final String SERVICE_KEY = "";
+
```
상수와 필드 변수와 구분해주세요~

drunkenhw reviewed (changes requested):
같이 페어로 했지만 혹시 뭔가 같이 실수 했을까봐 다시 확인해봤습니다~ 내일 같이 얘기해봐요

sosow0212 committed:
test: 충전기 및 충전소 업데이트 기능 테스트

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
test: 충전기 및 충전소 업데이트 기능 테스트

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
refactor: address 필드를 Charger가 아닌 ChargeStation이 가지도록 변경

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
refactor: address 필드를 Charger가 아닌 ChargeStation이 가지도록 변경

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
refactor: 팀 컨벤션에 맞게 코드 컨벤션 수정

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

sosow0212 committed:
refactor: 팀 컨벤션에 맞게 코드 컨벤션 수정

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateService.java:
```diff
@@ -0,0 +1,112 @@
+        List<ChargeStation> updatedStations = updateScrapper.updateData();
+
+        Map<String, ChargeStation> savedStationsByStationId = chargeStations.stream()
+                .collect(Collectors.toMap(ChargeStation::getStationId, it -> it));
```
Function.identity() 를 사용해도 좋을 것 같아요 it-> it 대신

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateService.java:
```diff
@@ -0,0 +1,112 @@
+    private void updateAllChargers(List<Charger> toUpdateChargers) {
+        if (!toUpdateChargers.isEmpty()) {
+            chargeUpdateJdbc.updateAllChargersBatch(toUpdateChargers);
+        }
```
isEmpty 체크가 jdbc 쪽으로 가면 좋을 것 같아요
service 레벨에서 알 필요는 없을테니까요

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateService.java:
```diff
@@ -0,0 +1,112 @@
+
+    private final UpdateScrapper updateScrapper;
+    private final ChargeStationRepository chargeStationRepository;
+    private final ChargeUpdateJdbc chargeUpdateJdbc;
```
JDBC 라는 구현체를 직접 알 필요는 없을 것 같아요
인터페이스를 한 단계 둬서, 추상화 하면 좋을 것 같아요

feb-dain committed:
feat: 버튼 스타일 컴포넌트를 구현한다 (#109)

* feat: 버튼 스타일 컴포넌트 구현

[#107]

* test: 설명 및 옵션 추가

[#107]

* feat: borderRadius에 `all` 옵션 추가

[#107]

-

Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>

kyw0716 committed:
refactor: google map을 전달하는 구조를 개선한다 (#111)

* refactor: 구글 지도 최초 로드 방식 수정

[#110]

* refactor: 구글 지도 사용 방식 수정

- 모든 컴포넌트에서 전역 상태에서 받아서 사용하는 방식으로 통일

[#110]

* refactor: 혼동될 수 있는 네이밍 수정

[#110]

* refactor: 혼동될 수 있는 네이밍 수정

[#110]

feb-dain committed:
test: 필요없는 파일을 제거하고 스토리북 기본 형식을 변경한다 (#117)

* test: 필요없는 파일(스토리북 관련) 제거

[#115]

* test: 스토리북 기본 형식 변경

[#115]

* test: 필요없는 코드 제거

[#115]

gabrielyoon7 committed:
feat: 현위치 버튼 클릭 시 현위치로 복귀하는 기능을 구현한다 (#118)

* feat: 아무 기능이 없는 현위치 복귀 버튼을 추가

* feat: 현위치 버튼을 누르면 현 위치로 이동하는 기능을 구현

kyw0716 committed:
chore: 마커 커스텀을 위한 svg import 환경을 설정한다 (#122)

be-student reviewed (approved):
이 부분만 바꿔주세요 👍 
고생 많으셨어요

kiarakim commented on backend/src/main/java/com/carffeine/carffeine/domain/chargestation/Longitude.java:
```diff
@@ -52,10 +52,11 @@ public boolean equals(Object o) {
         }
-        if (o == null || getClass() != o.getClass()) {
+
+        if (!(o instanceof Longitude longitude)) {
             return false;
         }
```
instanceof 보다 객체에게 직접 물어보는 것도 고려해보면 좋을 것 같아요. 
instanceof는 컴파일시, 객체가 사용할 instanceof()는 런타임에 타입 에러를 체크한다고 하네요.

feb-dain committed:
chore: 커밋시 프리티어와 eslint 적용을 자동화한다 (#126)

* test: 필요없는 파일 및 코드 삭제

[#114]

* chore: husky, lint-staged 설치 및 적용

[#103]

* chore: eslint fix 자동화

[#103]

kiarakim commented on backend/src/main/java/com/carffeine/carffeine/service/chargerstation/UpdateScrapper.java:
```diff
@@ -0,0 +1,70 @@
+
+            List<Charger> chargers = chargersByStationId.getOrDefault(chargeStationId, new ArrayList<>());
+
+            chargers.add(chargeStation.getChargers().get(0));
```
여기 get(0)의 0이 어떤 것을 의미하는지 상수화하주면 좋을 것 같아요

kiarakim commented on backend/src/test/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateServiceTest.java:
```diff
@@ -0,0 +1,119 @@
+        chargerStationUpdateService.updateStations();
+
+        // then
+        assertAll(
+                () -> verify(chargeUpdateJdbc, times(1)).saveAllStationsBatch(any()),
+                () -> verify(chargeUpdateJdbc, times(0)).updateAllStationsBatch(any()),
+                () -> verify(chargeUpdateJdbc, times(0)).saveAllChargersBatch(any()),
+                () -> verify(chargeUpdateJdbc, times(0)).updateAllChargersBatch(any())
+        );
```
여기선 assertAll 사용하고 ChargeUpdateJdpcTest에선 assertSoftly 사용하셨는데 통합하는 것은 어떨까유

kiarakim reviewed (changes requested):
instanceof는 꼭 바꾸라는 뜻은 아니고 comment인데 한번에 리뷰하니까 request changes로 리뷰할게요! 수고하셨습니다

kyw0716 committed:
feat: 충전소 마커를 필터링 하는 기능을 구현한다 (#124)

* fix: SetStateCallback 타입 추가 반영

[#123]

* test: 사용 가능한 충전기 개수 0개도 포함하도록 수정

[#123]

* feat: 충전소 필터링 옵션 전역 상태 추가

[#123]

* feat: 필터링 옵션 선택 컴포넌트 추가

[#123]

* feat: 충전소 정보 필터링 기능 추가

[#123]

* refactor: 직관적인 네이밍 적용

[#123]

* refactor: 직관적인 네이밍 적용

[#123]

feb-dain committed:
fix: husky에 `commit-msg` 추가 (#133)

[#131]

kyw0716 committed:
feat: 간단 정보 띄우는 기능 구현 (#132)

[#125]

gabrielyoon7 committed:
feat: 마지막 접속한 이력이 있는 클라이언트라면 마지막 위치를 우선하여 지도를 로드하는 기능을 추가한다 (#134)

* feat: 마지막 접속한 이력이 있는 클라이언트라면 마지막 위치를 우선하여 지도를 로드하는 기능 추가

[#130]

* refactor: 불필요한 import 제거

[#130]

feb-dain committed:
fix: mac 전용 prepare script 추가 (#137)

[#136]

kyw0716 committed:
fix: 웹팩 빌드 오류를 해결한다 (#138)

* chore: 웹팩 빌드 오류 수정

- styled-components의 createGlobalStyle 내부에 @import 삭제

[#135]

* refactor: 개발 환경에서만 msw가 동작하도록 분기처리한 부분 삭제

- 백엔드와 연결 성공 시 다시 분기 처리할 것

[#135]

drunkenhw committed:
fix: 위도, 경도 기반으로 조회 시 위도가 잘못되는 버그를 해결한다  (#141)

* fix: 계산 로직을 수정한다

[#139]

* refactor: entity graph를 활용해서 발생하는 쿼리를 줄인다

[#139]

kiarakim committed:
feat: 충전기 상태 및 혼잡도를 저장하는 기능을 구현한다 (#113)

* feat: 충전기 상태 업데이트 기능

[#21]

* feat: 주기적으로 충전기 정보 업데이트 기능 구현

[#21]

* feat: 혼잡도 정보 업데이트 기능 구현

[#21]

* refactor: 사용하지 않는 메서드 제거 및 매직넘버 처리

* refactor: 코드리뷰를 반영한다

[#21]

* feat: 혼잡도를 조회하는 컨트롤러 작성

[#21]

[#21)]

* feat: adoc 작성

[#21]

* feat: 처음 저장시 status 도 함께 저장하도록 변경

[#21]

* fix: charger 조회시, status null 인 경우 false를 반환하도록 한다

[#21]

---------

Co-authored-by: be-student <presentsong@naver.com>

drunkenhw committed:
refactor: 충전소를 업데이트 기능 메서드 병합

[#26]

drunkenhw committed:
Merge remote-tracking branch 'origin/feat/26' into feat/26

# Conflicts:
#	backend/src/main/java/com/carffeine/carffeine/domain/chargestation/ChargeStation.java
#	backend/src/main/java/com/carffeine/carffeine/domain/chargestation/charger/Charger.java
#	backend/src/main/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateService.java
#	backend/src/test/java/com/carffeine/carffeine/service/chargerstation/ChargerStationUpdateServiceTest.java

[#26]
